### ISSUE TITLE
Cross compilation produces:  Error running link command: No such file or directory

### ISSUE THREAD

**Issue by alaa137** [opencv/opencv#7594]
##### System information (version)
- OpenCV => 3.1
- Operating System (host) => Ubuntu 16.04.1
- Target platform => arm imx6 linux
- Compiler => arm-linux-gnueabi

##### Detailed description
Cross compiling OpenCV produces error: **Error running link command: No such file or directory**

##### Steps to reproduce
1. Clone latest openCV source from github: ```git clone https://github.com/opencv/opencv.git```
2. Create a build directory: ```mkdir /opencv/platforms/linux/arm_build```
3. Run ```cmake``` inside the build directory with arm compilers specified:  
```CC=arm-linux-gnueabihf-gcc CXX=arm-linux-gnueabihf-g++ cmake -DCMAKE_TOOLCHAIN_FILE=../arm-gnueabi.toolchain.cmake ../../..```
4. Run ```make```
5. Notice error on linking zlib:
```
Linking C static library ../lib/libzlib.a
Error running link command: No such file or directory
make[2]: *** [3rdparty/lib/libzlib.a] Error 2
make[1]: *** [3rdparty/zlib/CMakeFiles/zlib.dir/all] Error 2
make: *** [all] Error 2
```

##### Additional information  
- This error happens also if you run each 3rdparty library's make file separately.
- Running ```make VERBOSE=1``` produces:  
```CMAKE_AR-NOTFOUND qc ../lib/libzlib.a```

**Comment by alaa137** [comment]
**I found a workaround:**

After running cmake and generating the make files, edit `opencv/opencv/platforms/linux/arm_build/CMakeFiles/3.5.1/CMakeCCompiler.cmake` and replace `set(CMAKE_AR "CMAKE_AR-NOTFOUND")` with `set(CMAKE_AR "/usr/bin/arm-linux-gnueabi-ar")`

Note that setting the variable `CMAKE_AR` in the `cmake` line (`CMAKE_AR="/usr/bin/arm-linux-gnuabi-ar"`) does not work. Seems like this sets the flag to empty.

**Comment by alalek** [comment]
Fixed in #7645
